plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs

```diff
@@ -204,6 +204,18 @@ instance HasFromOpaque BuiltinBool Bool where
     fromOpaque b = ifThenElse b True False
     {-# INLINABLE fromOpaque #-}
 
+instance HasToOpaque [BuiltinInteger] (BuiltinList BuiltinInteger) where
```

Comment:
Why not `instance HasToOpaque a o => HasToOpaque [a] (BuiltinList o)`?

Reply:
I wanted to make it that, but it's pretty hard! We need to have the fully specialized `a` and compute the right type tag from it and it seems like it's above my Template Haskell skills currently. I can probably make it work without TH, but it won't be reliable. In any case, I do intend to create a follow-up polishing this mess a bit, this PR was just for fixing the immediate issue.